lib/srv/db/mysql/proxy.go

```diff
@@ -61,6 +61,10 @@ func (p *Proxy) HandleConnection(ctx context.Context, clientConn net.Conn) (err
 	// If any error happens, make sure to send it back to the client so it
 	// has a chance to close the connection from its side.
 	defer func() {
+		if r := recover(); r != nil {
+			p.Log.Warnf("Recovered in MySQL proxy while handling connection from %v: %v.", clientConn.RemoteAddr(), r)
+			err = trace.BadParameter("failed to handle MySQL client connection")
```

Comment:
Nit: can we get additional fields about the client such as source IP etc? So if someone is targetting us we can use the logs to identify the source?

Reply:
@knisbet The log entry includes client remote addr but that's probably not very useful if there's a load balancer. We also do log proxy line if lb sends it: https://github.com/gravitational/teleport/blob/12063b2cfe6c190823a4d4b9c8733ed3a0630821/lib/srv/db/mysql/proxy.go#L175

Would that work? I could also see if we can propagate the source client info from proxy line to this handler if that's better.

Reply:
Yea, that's a good point. I think the proxy-line would work, as we send proxy-protocol from our edge to the proxies.